addons/a11y/src/index.js

```diff
@@ -1,18 +1,35 @@
+import { document, setTimeout } from 'global';
+import axe from 'axe-core';
 import addons from '@storybook/addons';
+import Events from '@storybook/core-events';
+import { logger } from '@storybook/client-logger';
 
-import A11yManager from './A11yManager';
-import * as shared from './shared';
+import { CHECK_EVENT_ID, RERUN_EVENT_ID } from './shared';
 
-const manager = new A11yManager();
 let axeOptions = {};
 
-function checkA11y(storyFn, context) {
+export const configureA11y = (options = {}) => {
+  axeOptions = options;
+};
+
+const runA11yCheck = () => {
   const channel = addons.getChannel();
-  return manager.wrapStory(channel, storyFn, context, axeOptions);
-}
+  const wrapper = document.getElementById('root');
```

Comment:
Is the pattern we want to encourage? 

I feel like it is more inflexible than we might like, i.e it means stories can only ever be rendered in `#root`, whereas there are other conceivable ways you might want to render a story.

Could we instead do something like:

```js
import { afterStory } from '@storybook/core';

export const checkA11y = story => {
  addons.getChannel().emit(Events.REGISTER_SUBSCRIPTION, a11ySubscription);

  // The idea is that `afterStory` wraps a story in a callback that triggers with the story's 
  // DOM element after the story has rendered to the screen.
  const wrappedStory = afterStory(story, (html) => {
    runA11yCheck(html);
  });
 
 return wrappedStory();
};
```

Not sure if that's the best API, but does the idea make sense?

Reply:
Makes sense, will try something like `STORY_RENDERED` event

Not sure that I get what you mean by other ways to render a story though

Reply:
I guess what I mean is that at the moment you can call `const story = getStory()` on the story store and then execute that function to get something renderable (how you render that depends on the view layer of course). At the moment that function is also wrapped in the relevant decorators and those decorators will work however the story is rendered.

At the moment the only place a story is rendered is to the `#root` via the `renderMain` function, but other ways you might want to render a story is e.g. inside some `<Story/>` element in a documentation site.

Maybe I'm overthinking it, we would need to think about how decorators/addons make sense in that context a bit anyway, but it's something I think we will want to do in the future.

Reply:
I think it's a pattern we could support in [SBNext](https://github.com/storybooks/SBNext)

CC @ndelangen

Reply:
I've added `STORY_RENDERED` event to avoid `setTimeout(fn, 0)`, but I decided to keep `#root` part for now. Most of the addons, including a11y, have manager part, so they are unusable outside of storybook context. So I think that getting `rootEl` dynamically would bring more complexity than value right now

Reply:
WFM, thanks for talking through it with me.